app/models/collection_item.rb

```diff
@@ -34,6 +34,13 @@ def collection_is_open
     end
   end
 
+  validate :creators_allow_invitation, on: :create
+  def creators_allow_invitation
+    return unless item.respond_to?(:allow_collection_invitation)
+
+    errors.add(:base, :creator_invitation_not_allowed) unless item.allow_collection_invitation
+  end
```

Comment:
I don't think it's a good idea to put this check in the model, particularly with the way you've defined `Work#allow_collection_invitation`.

Currently, works can have collection items created in two different controllers: the `CollectionItemsController` (the "Add to Collection" form), and the `WorksController` (the "Edit Work" form). From the description of the issue, I think the intention is that the behavior in the "Edit Work" form should be unchanged, and this PR should affect only the behavior of the `CollectionItemsController`:
> When none of the creators of a work has checked "Allow others to invite my works to collections.", the "Add To Collections" button should be removed from the work. (**Creators can always edit the work itself to add it to a collection.**) 

(Emphasis mine.)

Reply:
You know, I think my brain just skipped over that parenthetical completely... thanks for calling me out and the push in the right direction!